File: ABC/117/c.java
import java.util.Arrays;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int numPieces = scanner.nextInt();
        int numTargets = scanner.nextInt();

        int[] targets = new int[numTargets];
        for (int i = 0; i < numTargets; ++i) {
            targets[i] = scanner.nextInt();
        }

        Arrays.sort(targets);

        int[] distances = new int[numTargets - 1];
        for (int i = 0; i < numTargets - 1; ++i) {
            distances[i] = targets[i + 1] - targets[i];
        }

        Arrays.sort(distances);

        long minMoves = 0;
        for (int i = 0; i < numTargets - numPieces; ++i) {
            minMoves += distances[i];
        }

        System.out.println(minMoves);
    }
}

File: ABC/117/c.kt
import java.util.Scanner
import java.util.Arrays

fun main(args: Array<String>) {
    val scanner = Scanner(System.`in`)

    val numPieces = scanner.nextInt()
    val numTargets = scanner.nextInt()

    val targets = IntArray(numTargets)
    for (i in 0 until numTargets) {
        targets[i] = scanner.nextInt()
    }

    Arrays.sort(targets)

    val distances = IntArray(numTargets - 1)
    for (i in 0 until numTargets - 1) {
        distances[i] = targets[i + 1] - targets[i]
    }

    Arrays.sort(distances)

    var minMoves: Long = 0
    for (i in 0 until numTargets - numPieces) {
        minMoves += distances[i].toLong()
    }

    println(minMoves)
}
